languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -10084,6 +10084,378 @@ though, and has since been slightly modified)-->
+              <antipattern>
+                  <token postag="SENT_START"/>
+                  <token regexp="yes" min="1" max="3">[^,);:(«»-]+</token>
+                  <token regexp="yes">mais|car</token>
+              </antipattern>
+              <antipattern>
+                  <token regexp="yes">[^,);:(«»-]+</token>
+                  <token regexp="yes">mais|car</token>
+                  <token regexp="yes" min="1" max="3">[^,);:(«»-]+</token>
+                  <token postag="SENT_END"/>
+              </antipattern>
+              <pattern>
+                  <marker>
+                      <token regexp="yes">[^,);:(«»-]+</token>
+                      <token regexp="yes">mais|car</token>
+                  </marker>
+              </pattern>
+              <message>Une virgule semble être requise dans ce contexte.</message>
+              <suggestion>\1, \2</suggestion>
+              <example correction="fâchée, mais">Depuis qu'il me l'a dit, je suis <marker>fâchée mais</marker> je ne lui en veux pas.</example>
+              <example>Il est <marker>beau mais</marker> grand.</example>
+          </rule>
+          <rule>
+              <pattern>
+                  <marker>
+                      <token regexp="yes">[^,);:(«»-]+</token>
+                      <token regexp="yes">voire</token>
+                  </marker>
+              </pattern>
+              <message>Une virgule semble être requise dans ce contexte.</message>
+              <suggestion>\1, \2</suggestion>
+              <example correction="fâchée, voire">Je suis <marker>fâchée voire</marker> furieuse !</example>
+          </rule>
+          <rule>
+              <pattern>
+                  <marker>
+                      <token regexp="yes">or</token>
+                      <token postag="R.* pers.* suj.*" postag_regexp="yes"/>
+                  </marker>
+              </pattern>
+              <message>Une virgule semble être requise dans ce contexte.</message>
+              <suggestion>\1, \2</suggestion>
+              <example correction="or, il">Il ne réagit pas au temps <marker>or il</marker> est pressé.</example>
+          </rule>
+          <rule>
+              <pattern>
+                  <marker>
+                      <token regexp="yes">[^,);:(«»-]+<exception postag="V.*|A.*" postag_regexp="yes"/><exception regexp="yes">peine|mal</exception></token>
+                      <token>à</token>
+                      <token>savoir</token>
+                  </marker>
+                  <token postag="D.*" postag_regexp="yes"/>
+              </pattern>
+              <message>Une virgule semble être requise dans ce contexte.</message>
+              <suggestion>\1, \2 \3</suggestion>
+              <example correction="famille, à savoir">Il viendra avec sa <marker>famille à savoir</marker> sa femme et ses deux enfants.</example>
+              <example><marker>Reste à savoir</marker> le prix.</example>
+          </rule>
+          <rule>
```

Comment:
Here, I'd just fix the thing with the quotes and other symbols like above, but that's it